identity/src/main.rs

```diff
@@ -1,59 +1,32 @@
-use std::{io, process, sync::Arc};
-
-use diesel::r2d2::{ConnectionManager, Pool};
-use dotenv::dotenv;
-use envconfig::Envconfig;
+use std::{io, sync::Arc};
 
 #[macro_use]
 extern crate diesel_migrations;
 
-use identity::config::Configuration;
-use identity::db::Db;
-use identity::rpc::rpc_server;
-
-static PKG_NAME: Option<&'static str> = option_env!("CARGO_PKG_NAME");
-static PKG_VERSION: Option<&'static str> = option_env!("CARGO_PKG_VERSION");
+use helpers::db::run_migration;
+use identity::{config::configuration, db::db_pool, rpc::rpc_server};
 
 #[tokio::main]
 async fn main() -> io::Result<()> {
     println!(
         "SERVICE: {} | VERSION: {}\n",
-        PKG_NAME.unwrap_or("<unknown>"),
-        PKG_VERSION.unwrap_or("<unknown>")
+        option_env!("CARGO_PKG_NAME").unwrap_or("<unknown>"),
+        option_env!("CARGO_PKG_VERSION").unwrap_or("<unknown>")
     );
     env_logger::init();
     log::info!("Starting service");
 
-    let configuration: Arc<Configuration> = {
-        dotenv().ok();
-        match Configuration::init_from_env() {
-            Ok(val) => Arc::new(val),
-            Err(e) => {
-                log::error!("{}", e);
-                log::error!("Terminating because of previous error.");
-                process::exit(1);
-            }
-        }
-    };
+    let configuration = Arc::new(configuration());
 
-    let db: Arc<Db> = {
-        match Pool::new(ConnectionManager::new(configuration.db_connection_url())) {
-            Ok(val) => Arc::new(val),
-            Err(e) => {
-                log::error!("{}", e);
-                log::error!("Terminating because of previous error.");
-                process::exit(1);
-            }
-        }
-    };
+    let db_pool = Arc::new(db_pool(&configuration.db_connection_url()));
 
     embed_migrations!();
-    helpers::db::run_migration(embedded_migrations::run, &db);
+    run_migration(embedded_migrations::run, &db_pool);
 
     let (server, addr) = rpc_server(
         configuration.rpc_socket(),
         configuration.clone(),
-        db.clone(),
+        db_pool.clone(),
```

Comment:
```suggestion
        Arc::new(db_pool),
```
In theory you could also not wrap the pool on initialization, but only on passing it to the `fn rpc_server(...)` and thereby don't need to call `.clone()`.

But just go with your preference :)